docs/source/reference/ha.rst

```diff
@@ -0,0 +1,345 @@
+|st2| deployment. This is perhaps the better of the two approaches to end up with a predicatble HA deployment of |st2|.
+
+Reference HA setup
+------------------
+
+In this section we provide a highly opinionated and therefore prescriptive approach to deploying |st2| in HA. This deployment
+has 3 independent boxes which we categorize as controller box and blueprint box. Let call these boxes ``st2-multi-node-cntl``,
+``st2-multi-node-1`` and ``st2-multi-node-2``. For the sake of reference we will be using Ubuntu 14.04 as the base operating
+system.
+
+.. figure :: /_static/images/st2-deployment-multi-node.png
+    :align: center
+
+    StackStorm HA reference deployment.
+
+.. source https://docs.google.com/drawings/d/1_BJa9ZtBjFa1Dxx6cPiFlmpTS9AsNzkkvp_vuyVV3bw/edit
+
+Controller box
+^^^^^^^^^^^^^^
+This box runs all the shared required dependencies and some |st2| components.
+
+* Nginx as loadbalancer
+* MongoDB
+* PostgreSQL
+* RabbitMQ
+* st2chatops
+* st2web
+
+In practice ``MongoDB``, ``PostgreSQL`` and ``RabbitMQ`` are often in standalone clusters managed outside of |st2|.
+The 2 shared components i.e. ``st2chatops`` and ``st2web`` are placed here for sake of convenience and could be placed anywhere
+with the right configuration.
+
+Nginx acting as the loadbalancer can easily be switched out for Amazon ELB, HAProxy or any other of your choosing. In that case
+``st2web`` which is being served off this Nginx will also need a new home.
+
+``st2chatops`` which use ``hubot`` is not easily deployed in HA. Using something like `keepalived <http://www.keepalived.org/>`__
+to maintain st2chatops in active-passive configuration would be an option.
+
+Following are the steps to provision a controller box on Ubuntu 14.04.
+
+Install required dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Install ``MongoDB``, ``PostgreSQL`` and ``RabbitMQ``.
+
+  .. code-block:: bash
+
+        sudo apt-get install -y mongodb-server rabbitmq-server postgresql
+
+
+2. Fix listen address in `/etc/postgresql/9.3/main/postgresql.conf` and have PostgreSQL listen an interface that has an
+   IP address reachable for ``st2-multi-node-1`` and ``st2-multi-node-2``.
+
+3. To `/etc/postgresql/9.3/main/pg_hba.conf` add ACL rule. Here the subnet to allow access is `10.0.3.1/24`.3
+
+  .. code-block:: bash
+
+        host       all  all  10.0.3.1/24  trust
+
+4. restart PostgreSQL
+
+  .. code-block:: bash
+
+         service postgresql restart
+
+5. Create Mistral DB in PostgreSQL.
+
+  .. code-block:: bash
+
+        cat << EHD | sudo -u postgres psql
+        CREATE ROLE mistral WITH CREATEDB LOGIN ENCRYPTED PASSWORD 'StackStorm';
+        CREATE DATABASE mistral OWNER mistral;
+        EHD
+
+6. Add stable |st2| repos.
+
+  .. code-block:: bash
+
+        curl -s https://packagecloud.io/install/repositories/StackStorm/staging-stable/script.deb.sh | sudo bash
+
+7. Setup st2web and SSL termination. Follow :ref:`install webui and setup ssl<ref-install-webui-ssl-deb>`.
```

Comment:
This part of the linked docs fails on the controller, since we didn't apt-get install st2.
```sudo cp /usr/share/doc/st2/conf/nginx/st2.conf /etc/nginx/sites-available/```

Could be helpful just to link to [the nginx config in the repo](https://github.com/StackStorm/st2/blob/master/conf/nginx/st2.conf).